torch/distributed/elastic/rendezvous/etcd_server.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import atexit
+import logging
+import os
+import shlex
+import shutil
+import socket
+import subprocess
+import tempfile
+import time
+from typing import Optional
+
+import etcd  # type: ignore[import]
+
+
+log = logging.getLogger(__name__)
+
+
+def find_free_port():
```

Comment:
(can be done in follow up PR), there is another `find_free_port` implementation in `common_utils.py`, do we need to consolidate?

https://github.com/pytorch/pytorch/blob/ac668c55e58e093270be2d72e17d801b83b97e45/torch/testing/_internal/common_utils.py#L1469

Reply:
ahh yes I did consider it. Thing is that the one in `common_utils` is under the `testing/_internal` directory and `etcd_server` is technically a part of the lib. I'll create a github issue for extracting this function into a proper util-lib module and having both use that.